src/Aspire.Dashboard/Components/Controls/SummaryDetailsView.razor.cs

```diff
@@ -21,20 +23,64 @@ public partial class SummaryDetailsView
     public required string DetailsTitle { get; set; }
 
     [Parameter]
-    public Orientation Orientation { get; set; } = Orientation.Horizontal;
+    public Orientation Orientation { get; set; } = Orientation.Vertical;
 
     [Parameter]
     public EventCallback OnDismiss { get; set; }
 
+    [Parameter]
+    public bool RememberSize { get; set; } = true;
+
+    [Parameter]
+    public bool RememberOrientation { get; set; } = true;
+
+    /// <summary>
+    /// Overrides the default key used to store the splitter size and orientation in local storage.
+    /// By default, the key is based on the current URL. If you have multiple instances of this control
+    /// on a page or want to share the same settings across multiple pages, you can set this property
+    /// </summary>
+    [Parameter]
+    public string? ViewKey { get; set; }
+
+    [Inject]
+    public required ProtectedLocalStorage ProtectedLocalStore { get; set; }
+
+    [Inject]
+    public required NavigationManager NavigationManager { get; set; }
+
     private readonly Icon _splitHorizontalIcon = new Icons.Regular.Size16.SplitHorizontal();
     private readonly Icon _splitVerticalIcon = new Icons.Regular.Size16.SplitVertical();
 
+    private string _panel1Size { get; set; } = "1fr";
+    private string _panel2Size { get; set; } = "1fr";
+
+    protected override async Task OnInitializedAsync()
+    {
+        if (RememberOrientation)
+        {
+            var orientationResult = await ProtectedLocalStore.GetAsync<Orientation>(GetOrientationStorageKey());
+            if (orientationResult.Success)
+            {
+                Orientation = orientationResult.Value;
+            }
+        }
+
+        if (RememberSize)
+        {
+            var panel1FractionResult = await ProtectedLocalStore.GetAsync<float>(GetSizeStorageKey());
+            if (panel1FractionResult.Success)
+            {
+                SetPanelSizes(panel1FractionResult.Value);
+            }
+        }
+    }
+
     private async Task HandleDismissAsync()
     {
         await OnDismiss.InvokeAsync();
     }
 
-    private void HandleToggleOrientation()
+    private async void HandleToggleOrientation()
```

Comment:
async Task

To be honest, I'm not sure what Blazor does with the task, but async void is rarely good.

Reply:
D'oh. Looks like I let VS add the `async` for me when I typed `await` but then there were no errors/warnings so I forgot to go back and change `void` to `Task` too. Thanks for the catch. Pushed a fix in #883